flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1171,6 +1174,38 @@ public TableResult executeInternal(Operation operation) {
                     .data(Collections.singletonList(Row.of(explanation)))
                     .setPrintStyle(TableResultImpl.PrintStyle.rawContent())
                     .build();
+        } else if (operation instanceof ExplainDetailOperation) {
+            ExplainDetailOperation explainOperation = (ExplainDetailOperation) operation;
+            ExplainDetail[] detailSet =
+                    explainOperation.getExplainDetails().stream()
+                            .map(
+                                    detail -> {
+                                        switch (detail) {
+                                            case "ESTIMATED_COST":
+                                                return ExplainDetail.ESTIMATED_COST;
+                                            case "CHANGELOG_MODE":
+                                                return ExplainDetail.CHANGELOG_MODE;
+                                            case "JSON_EXECUTION_PLAN":
+                                                return ExplainDetail.JSON_EXECUTION_PLAN;
+                                            default:
+                                                return null;
+                                        }
+                                    })
+                            .filter(Objects::nonNull)
+                            .distinct()
```

Comment:
Why need `distinct`? We have the promise in the validation phase that the details is distinct.